File: backend/airport-application-project/src/main/java/com/project/application/airportapplicationproject/entities/Person.java
package com.project.application.airportapplicationproject.entities;

import javax.persistence.*;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.*;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.io.Serializable;

@Entity
@Data
@EntityListeners(AuditingEntityListener.class)
public class Person implements Serializable {

	@Id
	@Setter(AccessLevel.NONE)
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	private Long id;

	@NotBlank(message = "Please specify the name")
	private String name;

	@NotBlank(message = "Please specify the surname")
	private String surname;

	@NotBlank(message = "Please specify the id card number")
	private String idCardNumber;

	@NotBlank(message = "Please specify the city")
	private String city;

	@Email(message = "Please enter data in the email form")
	@NotBlank(message = "Please specify the email")
	private String email;

	@Size(min = 9, max = 11, message = "Phone number should be between 9 and 11 character length")
	@NotBlank(message = "Please specify the phone number")
	private String phoneNumber;
}

File: frontend/airport/src/models/Person.ts
export class Person {
  id: number;
  name: string;
  surname: string;
  idCardNumber: string;
  city: string;
  email: string;
  phoneNumber: string;
  constructor(id: number, name: string, surname: string, idCardNumber: string,
              city: string, email: string, phoneNumber: string) {
    this.id = id;
    this.name = name;
    this.surname = surname;
    this.idCardNumber = idCardNumber;
    this.city = city;
    this.email = email;
    this.phoneNumber = phoneNumber;
  }
}
